src/sortStudents.ts

```diff
@@ -1,16 +1,80 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function averageGrades(grades: number[]): number {
+  const gradesSum = grades.reduce((sum, grade) => sum + grade, 0);
+  const averageGrade = gradesSum / (grades.length || 1);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return averageGrade;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const resultOfSortedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return order === 'asc'
+        ? resultOfSortedStudents.sort((studentA, studentB) => studentA.name
+          .localeCompare(studentB.name))
+        : resultOfSortedStudents.sort((studentA, studentB) => studentB.name
+          .localeCompare(studentA.name));
+
+    case SortType.Surname:
+      return order === 'asc'
+        ? resultOfSortedStudents.sort((studentA, studentB) => studentA.surname
+          .localeCompare(studentB.surname))
+        : resultOfSortedStudents.sort((studentA, studentB) => studentB.surname
+          .localeCompare(studentA.surname));
```

Comment:
we can combine some cases (you can do the same with age and married)

```suggestion
    case SortType.Name:
    case SortType.Surname:
      return order === 'asc'
        ? resultOfSortedStudents.sort((studentA, studentB) => studentA[sortBy]
          .localeCompare(studentB[sortBy]))
        : resultOfSortedStudents.sort((studentA, studentB) => studentB[sortBy]
          .localeCompare(studentA[sortBy]));